### PR TITLE
AES-128-CBC: allow 16 bytes of padding

### PR DESCRIPTION
**Pull request by fialhopm**
Allow 16 bytes of padding when decoding an AES-128-CBC confidentiality layer. This was discussed in [this](https://github.com/gebn/bmc/issues/62#issuecomment-1878479915) issue.

### CODE REVIEW

File: pkg/ipmi/aes_128_cbc_test.go [hunk 1]
```diff
@@ -47,8 +47,30 @@ func TestAES128CBCDecodeFromBytes(t *testing.T) {
 				0x3a, 0x3c, 0x9b, 0x30, 0x81,
 			},
 		},
-		// 16 bytes data (must flow onto second block for padding length, max
-		// padding)
+		// 15 bytes of data encrypted with 16 bytes of padding.
+		// This is not how AES128CBC.SerializeTo would encrypt this message, but
+		// other implementations of AES in CBC mode may work this way (e.g.
+		// OpenSSL).
+		{
```
Comment (fialhopm) [inline pkg/ipmi/aes_128_cbc_test.go:54]: This test fails with `unexpected error: invalid number of pad bytes: 16` if we remove the `pkg/ipmi/aes_128_cbc.go` change.

**Comment by gebn** [comment]
Thank you! Looks great, just two trivial comments then happy to merge.

**Comment by fialhopm** [comment]
Thanks for taking a look! I don't see the two comments though.

**Comment by gebn** [comment]
Helps if I submit the review... 🤦‍♂️

File: pkg/ipmi/aes_128_cbc.go [hunk 2]
```diff
@@ -65,8 +65,12 @@ func (a *AES128CBC) DecodeFromBytes(data []byte, _ gopacket.DecodeFeedback) erro
 	// ciphertext is useless on its own, so we don't mind
 	mode.CryptBlocks(data[a.cipher.BlockSize():], data[a.cipher.BlockSize():])
 
-	padBytes := uint8(data[len(data)-1])
-	if padBytes > uint8(a.cipher.BlockSize()-1) {
+	padBytes := data[len(data)-1]
```
[the remark this answers is not in the document]
Comment (fialhopm) [reply pkg/ipmi/aes_128_cbc.go]: Sounds good. I only changed it because GoLand tagged it as redundant.

[the remark this answers is not in the document]
Comment (gebn) [reply pkg/ipmi/aes_128_cbc.go]: I'm surprised at JetBrains! `byte` is indeed an alias of `uint8` in Go, but they have very different semantic meaning. Here it's definitely an integer as opposed to 8 1s or 0s.

**Pull request merged**